[splash-screen] add splash-screen-hanging.md (#31)
expo/fyi: splash-screen-hanging.md

# Stuck on the Splash Screen?

> If you've recently encountered this popup while developing your app, chances are something has gone wrong with the SplashScreen component. Here are a couple of possible reasons why you might be seeing this with your app: 

Reviewer: ```suggestion
It can be frustrating to work on your app when you can't even get it to load past the splash screen. Here are a couple of possible reasons why you might be seeing this with your app:
```

> ### 1. SplashScreen.hideAsync() was never called

Reviewer: ```suggestion
### 1. You may have forgotten to call `SplashScreen.hideAsync()`
```

> If you have been leveraging the SplashScreen component to preload extra data and assets while your app starts up, then it's possible you simply forgot to call this method! 

Reviewer: ```suggestion
If you call `SplashScreen.preventAutoHideAsync()` to do some work while the splash screen is displayed (for example, to preload extra data and assets while your app starts up), you need to ensure that you always end up calling `SplashScreen.hideAsync()`. It's possible that you forgot to call this method!
```

> ### 2. Unresolved promise on app startup 

Reviewer: ```suggestion
### 2. An error may have prevented `SplashScreen.hideAsync()` from being called
```

> If you have an unresolved promise that prevents `hideAsync()` from being called, then you'll be stuck on the splash screen indefinitely. You can verify what is going wrong by checking the terminal logs: 

Reviewer: ````suggestion
Let's say you structure your code like this:

```js
async function loadResourcesAsync() {
  await loadStoredDataAsync();
  await loadFontsAsync();
  SplashScreen.hideAsync();
}
```

If either `await loadStoredDataAsync()` or `await loadFontsAsync()` throw an exception, you will never reach the `SplashScreen.hideAsync()` call.

In other words, if you have an unresolved promise or anything that will prevent `hideAsync()` from being called, then you'll be stuck on the splash screen indefinitely. Often, you can verify what is going wrong by checking the terminal logs: 
````

<img src="./assets/splash-screen-hanging/error-console.png" width="300" alt="Error Console" />

> 

Reviewer: ### Still having trouble?

Refer to the ["Manual debugging" guide](https://expo.fyi/manual-debugging) to learn more about how you can go about narrowing down the root cause of the problem in your unique circumstances.
